include/uapi/sound/sof/tokens.h

```diff
@@ -88,6 +88,8 @@
 #define SOF_TKN_COMP_CPC			406
 #define SOF_TKN_COMP_IS_PAGES			409
 #define SOF_TKN_COMP_NUM_AUDIO_FORMATS		410
+#define SOF_TKN_COMP_NUM_SOURCE_PINS		411
```

Comment:
I'd simplify the commit message to "Add support for parsing the number of source/sink pins per widget from topology. These will be used to determine the source/sink queue ID's during widget binding"